lib/cli/history.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+
+const historyPath = process.env.JSTP_CLI_HISTORY;
+const historyDisabled = historyPath === '';
+const historySize = Number(process.env.JSTP_CLI_HISTORY_SIZE);
+const DEFAULT_HISTORY_SIZE = 1000;
+const DEFAULT_FILENAME = '.jstp_cli_history';
+
+const getHistorySize = () => {
+  if (historyDisabled) return 0;
+  return Number.isNaN(historySize) || historySize <= 0 ?
+    DEFAULT_HISTORY_SIZE :
+    historySize;
+};
+
+const historyHandle = Symbol('historyHandle');
+
+const setupHistory = (rl, callback) => {
+  if (historyDisabled) {
+    callback();
+    return;
+  }
+  let resolvedPath;
+  try {
+    resolvedPath = historyPath || path.join(os.homedir(), DEFAULT_FILENAME);
+  } catch (err) {
+    callback(err,
+      'Error: could not get home directory\nHistory will not be persisted');
+    return;
+  }
+
+  createFile();
+
+  function createFile() {
+    fs.open(resolvedPath, 'a+', 0o0600, (err, fd) => {
+      if (err) {
+        callback(err,
+          'Error: could not open history file\nHistory will not be persisted');
+        return;
+      }
+      fs.close(fd, onClose);
```

Comment:
Why close and then reopen in `loadHistory` if we could just read from this `fd`?